x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/StoreKeyConfigTests.java

```diff
@@ -52,6 +53,42 @@ public void testKeyStorePathCanBeEmptyForPkcs11() throws Exception {
         assertThat(ee.getCause().getMessage(), containsString("PKCS11 not found"));
     }
 
+    public void testCreateKeyManagerFromPKCS12ContainingCA() throws Exception {
+        assumeFalse("Can't run in a FIPS JVM", inFipsJvm());
+        final Settings settings = Settings.builder().put("path.home", createTempDir()).build();
+        final String path = getDataPath("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/httpCa.p12").toString();
+        final SecureString keyStorePassword = new SecureString("password".toCharArray());
+        final StoreKeyConfig keyConfig = new StoreKeyConfig(path, "PKCS12", keyStorePassword, keyStorePassword,
+            KeyManagerFactory.getDefaultAlgorithm(), TrustManagerFactory.getDefaultAlgorithm());
+        final X509ExtendedKeyManager keyManager = keyConfig.createKeyManager(TestEnvironment.newEnvironment(settings));
+        final PrivateKey ca_key = keyManager.getPrivateKey("cakey");
+        final PrivateKey http_key = keyManager.getPrivateKey("http");
+        final String[] aliases = keyManager.getServerAliases("RSA", null);
+        final X509Certificate[] certificates = keyManager.getCertificateChain("http");
+        assertThat(ca_key, equalTo(null));
+        assertThat(http_key, notNullValue());
+        assertThat(aliases.length, equalTo(1));
+        assertThat(aliases[0], equalTo("http"));
+        assertThat(certificates.length, equalTo(2));
```

Comment:
I think we can remove this assertion, it doesn't test something that we are interested in ( the certificate chain length of the non-ca certificate we happened to use in the test )

Reply:
Just making sure we haven't delete something we didn't mean to (like it happened before the last chage)

Reply:
This the certificate chain length of a single certificate chain of a single PrivateKeyEntry. ( `final X509Certificate[] certificates = keyManager.getCertificateChain("http");` ) 

We wouldn't have deleted one of the certificates in that chain before the last change.

Reply:
I still think we don't need this, but won't block the PR on this